test/explain/test_explainer.py

```diff
@@ -0,0 +1,280 @@
+
+
+class DummyModel(torch.nn.Module):
+    def __init__(self, out_dim: int = 1) -> None:
+        super().__init__()
+        self.out_dim = out_dim
+
+    def forward(self, g: Data, **kwargs):
+        return g.x.mean().repeat(self.out_dim)
+
+
+class DummyModelNotCompatible(torch.nn.Module):
+    def forward(self, x: torch.Tensor, **kwargs) -> torch.Tensor:
+        return x
+
+
+class DummyExplainer(ExplainerAlgorithm):
+    def forward(
+        self,
+        g: Data,
+        **kwargs,
+    ) -> Data:
+        return g
+
+    def explain(
+        self,
+        g: Data,
+        **kwargs,
+    ) -> Data:
+        return g
+
+    def supports(
+        self,
+        explanation_type: str,
+        mask_type: str,
+    ) -> bool:
+        return True
+
+    def loss(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        return torch.nn.functional.mse_loss(y_hat, y)
+
+
+@pytest.fixture(scope="module")
+def dummyexplainer():
+    return DummyExplainer()
+
+
+@pytest.fixture(scope="module")
+def dummyInput() -> Data:
+    return g
+
+
+@pytest.fixture(scope="module")
+def explanation_algorithm() -> ExplainerAlgorithm:
+    return RandomExplainer()
+
+
+models = [DummyModel(), DummyModel(2)]
+model_return_types = ModelConfig._valid_model_return_type
+explanation_types = ExplainerConfig._valid_explanation_type
+mask_types = ExplainerConfig._valid_mask_type
+thresholds = Threshold._valid_type
+task_levels = ModelConfig._valid_task
+
+
+@pytest.mark.parametrize("model", models[:1])
+@pytest.mark.parametrize("model_return_type", model_return_types[:1])
+@pytest.mark.parametrize("task_level", task_levels[:1])
+@pytest.mark.parametrize("explanation_type", explanation_types[:1])
+@pytest.mark.parametrize("mask_type", mask_types[:1])
+@pytest.mark.parametrize("threshold_pairs", [("none", 0, True),
+                                             ("hard", 0.5, True),
+                                             ("hard", 1.1, False),
+                                             ("hard", -1, False),
+                                             ("topk", 1, True),
+                                             ("topk", 0, False),
+                                             ("topk", -1, False),
+                                             ("topk", 0.5, False),
+                                             ("connected", 1, True),
+                                             ("connected", 0, False),
+                                             ("connected", -1, False),
+                                             ("connected", 0.5, False),
+                                             ("invalid", None, False)])
+def test_classification_threshold_init(dummyexplainer, model,
+                                       model_return_type, task_level,
+                                       explanation_type, mask_type,
+                                       threshold_pairs):
+    if threshold_pairs[2]:
```

Comment:
(nit) its pretty hard to tell what this test is trying to test: maybe it would be a bit clearer if we added a line
```
threshold, threshold_value, allowed = threshold_pairs
```